### PR TITLE
Add `--dry-run` to `remove-demo` command

### LINKED ISSUE #2117: Add `--dry-run` to remove-demo command
This doesn't need to be done before v8, but it could be good to add a `--dry-run` flag to the `ignite-cli remove-demo` command, where we just print out what would be modified and deleted.

### PR DESCRIPTION
**Pull request by jamonholmgren**
Fixes #2117.

CleanShot 2022-09-09 at 18 56 20@2x

### CODE REVIEW

**Comment by jamonholmgren** [comment]
btw @joshuayoes, I think we shouldn't display the full path, but rather the relative path of the files.

So instead of this:

```
   Found '@demo remove-file' in /Users/jh/Code/Temp/PizzaApp/app/models/AuthenticationStore.ts
   Found '@demo remove-file' in /Users/jh/Code/Temp/PizzaApp/app/models/Episode.test.ts
   Found '@demo remove-file' in /Users/jh/Code/Temp/PizzaApp/app/models/Episode.ts
```

it would be this:

```
   Found '@demo remove-file' in app/models/AuthenticationStore.ts
   Found '@demo remove-file' in app/models/Episode.test.ts
   Found '@demo remove-file' in app/models/Episode.ts
```

Less visual clutter that way.

File: src/commands/remove-demo.ts [hunk 1]
```diff
@@ -19,14 +19,17 @@ const MATCHING_GLOBS = [
...
   alias: ["rd", "remove-demos"],
-  description: "Remove demo code from generated boilerplate",
+  description:
+    "Remove demo code from generated boilerplate. Add --dry-run to see what would be removed.",
   run: async (toolbox: GluegunToolbox) => {
     const { parameters, patching, filesystem } = toolbox
 
     const CWD = process.cwd()
     const TARGET_DIR = parameters.first ?? CWD
+    const dryRun = parameters.options.dryRun ?? false
```
Comment (joshuayoes) [inline src/commands/remove-demo.ts:29]: Going to want to wrap this in a boolFlag, or else it will be `'true'`, `'false'`, or `undefined`

Comment (joshuayoes) [inline src/commands/remove-demo.ts:29]: Going to want to wrap this in a boolFlag, or else it will be `'true'`, `'false'`, or `undefined`

**Review by joshuayoes — COMMENTED** [review]
Looks good except for my one comment about wrapping `--dry-run` in a boolFlag!

**Pull request merged**